mantle/platform/qemu.go

```diff
@@ -306,6 +306,61 @@ func (inst *QemuInstance) SwitchBootOrder() (err error) {
 	return nil
 }
 
+// RemovePrimaryBlockDevice deletes the primary device from a qemu instance
+// and sets the seconday device as primary. It expects that all block devices
+// are mirrors.
+func (inst *QemuInstance) RemovePrimaryBlockDevice() error {
+	var primaryDevice string
+	var secondaryDevicePath string
+	monitor, err := newQMPMonitor(inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+	if err := monitor.Connect(); err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+
+	defer func() {
+		e := monitor.Disconnect()
+		if err != nil {
+			err = fmt.Errorf("%v; %v", err, e)
+		} else {
+			err = e
+		}
+	}()
+	blkdevs, err := listQMPBlkDevices(monitor, inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not list block devices through qmp")
+	}
+	for _, dev := range blkdevs.Return {
+		if !dev.Removable {
+			if dev.Inserted.BackingfileDepth == 1 {
```

Comment:
I agree with the point that we might need to add that functionality in the future but looking through the code it might be a bit hairy in practice to actually determine what information we need to plumb out of `AddFd` to be able to correlate it to the position we process it in the list of `builder.fds`. As of right now I believe the primary disk should always be the first fd in the list and as such we could make an assumption of `/dev/fdset/3` but there's nothing guaranteeing that into the future.

I suppose we could also check the file referenced in `.inserted.image.backing-image.filename` and compare that to the disk image but that also feels like something that isn't entirely future proof.

Reply:
Yeah, I agree it'd require some refactoring. Not worth blocking on this right now!
Though maybe for now a comment explaining what this is doing and an associated `XXX`?